app/channel.go

```diff
@@ -1570,6 +1578,16 @@ func (a *App) AutocompleteChannelsForSearch(teamId string, userId string, term s
 	return result.Data.(*model.ChannelList), nil
 }
 
+func (a *App) SearchAllChannels(term string) (*model.ChannelListWithTeamData, *model.AppError) {
+	includeDeleted := *a.Config().TeamSettings.ExperimentalViewArchivedChannels
```

Comment:
Does it make sense to include deleted channels in the response always if that config is set to on? I can't imagine you'd want to include deleted channels in the LDAP group-syncables target-picker ever?

Reply:
@jespino ^

Reply:
Yes, you are right, probably makes sense to have another parameter and use a get parameter to decide when to use and when not. This is used by the channel selector, probably here makes no sense to list them, but maybe, in other places would need archived channels.

Reply:
Perhaps we can just leave that out until it's actually needed for something?

Reply:
0/5 on this, btw.